docs/specifications/tests/Methods.md

```diff
@@ -35,25 +35,25 @@ This is an external method to be used by test cases.
 
 ### Ordered description of steps to be taken to execute the method
 
-1. Get the name servers set ("Name Servers") where each unique name 
-   server name is linked to a set of its possibly empty set of IP 
+1. Get the name server set ("Name Servers") where each unique name 
+   server name is linked to a possibly empty set of its IP 
    addresses by running [get-delegation].
 
 2. Extract the set of name server names ("Names") from 
    *Name Servers*.
 
-3. Get IP addresses for any [out-of-bailiwick] name server
+3. Get the IP addresses for any [out-of-bailiwick] name server
    names in *Names* by running [get-oob-ips] with *Names*
    as input.
 
 4. Merge the set returned with *Name Servers*.
 
-5. Return *Name Servers*.
+5. Output the *Name Servers*.
 
-### Outcome(s)
+### Outputs
```

Comment:
Regarding the placement of the Outputs section.

> For a human program I think it is better to have it after the steps. 

I wholeheartedly disagree. It makes no difference if programs are expressed in a proper programming language or in a human language. Programs are written to be understood by humans.

It's very common that you just want a general view of what a program does and don't care about the details of its operation. You a very good way of doing that is to look at the method's name, its description, its inputs and its outputs.

Also, when you're looking at the operation of a method, it's very helpful to know its outputs beforehand. It's not helpful to force the reader to scroll down to the output specification below the operational description (which is often by far the longest section).